spec/Data_Parallelism.tex

```diff
@@ -405,6 +419,32 @@ \section{Promotion}
 is performed with getter methods~(\rsec{Getter_Methods}), field
 access can also be promoted.
 
+If the original function returns a value, the corresponding promoted
+expression is an iterator yielding each computed value in turn.
```

Comment:
returns a value ie. this doesn't apply to functions returning a reference? Maybe you mean "if the original function does not return void" or "returns at all"?